foundry/applications/formApplications/baseEntitySheets/rollTableConfig.d.ts

```diff
@@ -1,25 +1,151 @@
+  static get defaultOptions(): RollTableConfig.Options;
+
   /**
-   * Return a dynamic application title for the configuration sheet
+   * @override
    */
   get title(): string;
 
   /**
-   * Activate event listeners for the RollTable Configuration Sheet
+   * @param options - (unused)
+   * @override
+   */
+  getData(options?: Application.RenderOptions): RollTableConfig.Data;
+
+  /**
+   * @override
    */
   activateListeners(html: JQuery): void;
 
   /**
-   * @param options - (unused)
+   * Handle creating a TableResult in the RollTable entity
+   * @param event      - The originating mouse event
+   * @param resultData - An optional object of result data to use
+   */
+  protected _onCreateResult(event: MouseEvent, resultData?: object): Promise<any>; // TODO: type when TableResult is typed
+
+  /**
+   * Submit the entire form when a table result type is changed, in case there are other active changes
+   */
+  protected _onChangeResultType(event: Event): ReturnType<RollTableConfig['_onSubmit']>;
+
+  /**
+   * Handle deleting a TableResult from the RollTable entity
+   */
+  protected _onDeleteResult(event: Event): Promise<any>; // TODO: type when TableResult is typed
+
+  /**
    * @override
    */
-  getData(options?: any): BaseEntitySheet.Data<Entity>;
+  protected _onDrop(event: DragEvent): ReturnType<RollTableConfig['_onCreateResult']>;
+
+  /**
+   * Handle changing the actor profile image by opening a FilePicker
+   */
+  protected _onEditImage(event: Event): ReturnType<FilePicker['browse']>;
+
+  /**
+   * Handle a button click to re-normalize dice result ranges across all RollTable results
+   */
+  protected _onNormalizeResults(event: Event): Promise<any>; // TODO: type when TableResult is typed
+
+  /**
+   * Handle toggling the drawn status of the result in the table
+   */
+  protected _onLockResult(event: Event): Promise<any>; // TODO: type when TableResult is typed
+
+  /**
+   * Reset the Table to it's original composition with all options unlocked
+   */
+  protected _onResetTable(event: Event): Promise<any>; // TODO: type when TableResult is typed
+
+  /**
+   * Handle drawing a result from the RollTable
+   */
+  protected _onRollTable(event: Event): Promise<void>;
+
+  /**
+   * Configure the update object workflow for the Roll Table configuration sheet
+   * Additional logic is needed here to reconstruct the results array from the editable fields on the sheet
+   * @param event    - The form submission event
+   * @param formData - The validated FormData translated into an Object for submission
+   */
+  protected _updateObject(event: Event, formData: RollTableConfig.Data): Promise<any>; // TODO: type when TableResult is typed
+
+  /**
+   * Display a roulette style animation when a Roll Table result is drawn from the sheet
+   * @param results - An Array of drawn table results to highlight
+   */
+  protected _animateRoll(results: object): Promise<void>; // TODO: type when TableResult is typed
```

Comment:
`results` should be of type `RollTable.Result[]`. And it seems like the return type should be `Promise<void[]>` (not that it matters much).